### ISSUE TITLE
[Question] What is the rationale behind adding the Prometheus annotations to the Kubernetes Service?

### ISSUE THREAD

**Issue by carlosjgp** [strimzi/strimzi-kafka-operator#2697]
Hi Strimzi team,

we are using Strimzi to deploy our Kafka Connect clusters
and we are very happy with the work you are doing :rocket: 

Yesterday we noticed we were ingesting the same metrics twice
because we added the Prometheus annotations to the pod template CRD
and the operator is creating the Kubernetes service with the same annotations
https://github.com/strimzi/strimzi-kafka-operator/blob/master/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java#L272

We prefer the POD metrics because they are more granular and using the Service makes Prometheus requests to be proxied randomly into the PODs (at least on a default EKS deployment)
https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-iptables

We had to exclude these services explicitly in our Prometheus configuration
which is not ideal.
```
                - job_name: 'kubernetes-service-endpoints'

                  kubernetes_sd_configs:
                    - role: endpoints

                  relabel_configs:
                    # KafkaConnect cluster deployed with Strimzi configures the Service with
                    # the Prometheus annotations so this job will scrape the Endpoint but
                    # instead of that we want to scrape each POD individually to get more granular data
                    - source_labels: [__meta_kubernetes_service_label_strimzi_io_kind]
                      action: drop
                      regex: KafkaConnect
                    # Add more exceptions here
...
```

I would like to understand what was the motivation to add these annotations here in case I'm missing something

Thanks

**Comment by scholzj** [comment]
I thin we added it in the past when the template was not available and people needed it. Maybe now it is time to remove it, although it is a bit of a breaking change? I'm not sure how much they are still needed TBH, I'm not really a Prometheus expert.

@ppatierno @stanlyDoge WDYT?

**Comment by scholzj** [comment]
@ppatierno @stanlyDoge ping

**Comment by stanlyDoge** [comment]
If you want to scrape metrics directly from the pods, so you can use `role: pod`, cannot you?

**Comment by scholzj** [comment]
I think the problem is that Strimzi does not live alone in the cluster. And as far as I understood it, the annotations which are there out of the box lead to the metrics being scrapedregardless whether you want or not - for example by wrong Prometheus instance.

**Comment by stanlyDoge** [comment]
So you think we should use different (non prometheus) annotations to make it work for us? It should be just matter of changing these "selectors" https://github.com/strimzi/strimzi-kafka-operator/blob/d218ab79e93c88db5cb12b3ccb232671f32b66ed/examples/metrics/prometheus-additional-properties/prometheus-additional.yaml#L183

**Comment by scholzj** [comment]
Maybe ... TBH I do not completely know what the options are. Isn't the `ServiceMonitor` or the `PodMonitor` already configuring how our metrics are scraped in our Prometheus deployment?

**Comment by stanlyDoge** [comment]
It seems `kubernetes-services` could be transformed from additional properties to `ServiceMonitor`.

**Comment by carlosjgp** [comment]
> If you want to scrape metrics directly from the pods, so you can use role: pod, cannot you?

It depends on how you configure Prometheus and the annotations added to Kafka connect `POD` or `Service`

The most common configuration for Prometheus deployments are

```
        kubernetes_sd_configs:
          - role: pod

        relabel_configs:
          - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
            action: keep
            regex: true
``` 

This block is common for `POD` and `Service`. Meaning that Prometheus will discover and scrape any URL (endpoint) configured with that annotation
```
        kubernetes_sd_configs:
          - role: endpoints

        relabel_configs:
          - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape]
            action: keep
            regex: true
```

Currently, that annotation is added to Strimzi Kafka Connect cluster `Service` that's why I have to explicitly ignore it on my Prometheus configuration and add the annotations to the POD template on the Strimzi Connect CRD

`ServiceMonitor` and `PodMonitor` are Prometheus Operator CRDs that will render Prometheus configuration specifically for a `Service` or `POD` without using the Prometheus annotations

------------------------------

My proposal is to give the control to the end-user to choose if they want to use Prometheus Operator CRDs, annotations and where to add the annotations to scrape the Kafka connect cluster metrics

**Comment by scholzj** [comment]
The prometheus annotation was deprecated in 0.19.0 and will be removed in 0.20.0: https://github.com/strimzi/strimzi-kafka-operator/blob/master/CHANGELOG.md#deprecation-of-monitoring-port-on-kafka-and-zookeeper-related-services

@ppatierno Will you work on it?

**Comment by ppatierno** [comment]
@scholzj yes I will do.

**Comment by carlosjgp** [comment]
I'll close this now since it will be resolved on `0.20.0` release
https://github.com/strimzi/strimzi-kafka-operator/pull/3439

Thanks!
